<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>723</width>
    <height>442</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Demo3_7 QAction,QMainWindow,QPlainTextEdit</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPlainTextEdit" name="plainTextEdit">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>10</y>
      <width>541</width>
      <height>341</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>from PyQt5.QtWidgets import QWidget,QApplication
from PyQt5.QtCore import QTime,QDate,QDateTime
import sys

from ui_Widget import Ui_Widget

class QmyWidget(QWidget):
    def __init__(self):
        super().__init__(parent = None)
        self.ui = Ui_Widget()
        self.ui.setupUi(self)
        print(dir(self.ui))
        self.ui.calendarWidget.selectionChanged.connect(lambda :print(self.ui.calendarWidget.selectedDate().toString('yyyy-MM-dd')))

    def on_calendarWidget_selectionChanged(self):
        self.ui.editCalender.setText(self.ui.calendarWidget.selectedDate().toString('yyyy年MM月dd日'))

    def on_btnSetTime_clicked(self):
        timestr = self.ui.editTime.text()
        time = QTime.fromString(timestr,'hh:mm:ss') #fromstring 将字符串格式时间转换为Qtime对象
        self.ui.timeEdit.setTime(time)
    def on_btnSetDate_clicked(self):
        date = self.ui.editDate.text()
        qdate = QDate.fromString(date,'yyyy-MM-dd')#fromstring 将字符串格式时间转换为Qdate对象
        self.ui.dateEdit.setDate(qdate)

    def on_btnSetDateTime_clicked(self):
        datetimeStr = self.ui.editDateTime.text()
        datetime = QDateTime.fromString(datetimeStr,'yyyy-MM-dd hh:mm:ss')#fromstring 将字符串格式时间转换为Qdatetime对象
        self.ui.dateTimeEdit.setDateTime(datetime)
    def on_btnGetTime_clicked(self):
        curDateTime = QDateTime.currentDateTime() #获取当前日期时间并返回一个datetime对象

        self.ui.timeEdit.setTime(curDateTime.time())
        print(curDateTime.time())
        self.ui.editTime.setText(curDateTime.time().toString('hh:mm:ss'))#tostring 将Qtime对象转换为字符串格式

        self.ui.dateEdit.setDate(curDateTime.date())
        self.ui.editDate.setText(curDateTime.date().toString('yyyy-MM-dd'))#tostring 将Qdate对象转换为字符串格式

        self.ui.dateTimeEdit.setDateTime(curDateTime)
        self.ui.editDateTime.setText(curDateTime.toString('yyyy-MM-dd hh:mm:ss'))#tostring 将Qdatetime对象转换为字符串格式


if __name__ == '__main__':
    app = QApplication(sys.argv)
    form = QmyWidget()
    form.show()
    sys.exit(app.exec_())</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>723</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件(F)</string>
    </property>
    <addaction name="actFile_New"/>
    <addaction name="actFile_Open"/>
    <addaction name="actFile_Save"/>
    <addaction name="separator"/>
    <addaction name="actClose"/>
   </widget>
   <widget class="QMenu" name="menu_E">
    <property name="title">
     <string>编辑(E)</string>
    </property>
    <addaction name="actEdit_Cut"/>
    <addaction name="actEdit_Copy"/>
    <addaction name="actEdit_Paste"/>
    <addaction name="separator"/>
    <addaction name="actEdit_Undo"/>
    <addaction name="actEdit_Redo"/>
    <addaction name="separator"/>
    <addaction name="actEdit_SelectAll"/>
    <addaction name="actEdit_Clear"/>
   </widget>
   <widget class="QMenu" name="menu_M">
    <property name="title">
     <string>格式(M)</string>
    </property>
    <widget class="QMenu" name="menu_2">
     <property name="title">
      <string>界面语言</string>
     </property>
     <addaction name="actLang_CN"/>
     <addaction name="actLang_EN"/>
    </widget>
    <addaction name="actFont_Bold"/>
    <addaction name="actFont_Italic"/>
    <addaction name="actFont_UnderLine"/>
    <addaction name="separator"/>
    <addaction name="actSys_ToggleText"/>
    <addaction name="menu_2"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_E"/>
   <addaction name="menu_M"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>true</bool>
   </attribute>
   <addaction name="actFile_New"/>
   <addaction name="actFile_Open"/>
   <addaction name="actFile_Save"/>
   <addaction name="separator"/>
   <addaction name="actEdit_Cut"/>
   <addaction name="actEdit_Copy"/>
   <addaction name="actEdit_Paste"/>
   <addaction name="separator"/>
   <addaction name="actEdit_Undo"/>
   <addaction name="actEdit_Redo"/>
   <addaction name="separator"/>
   <addaction name="actFont_Bold"/>
   <addaction name="actFont_Italic"/>
   <addaction name="actFont_UnderLine"/>
   <addaction name="separator"/>
   <addaction name="actLang_CN"/>
   <addaction name="actLang_EN"/>
   <addaction name="separator"/>
  </widget>
  <action name="actFile_New">
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/100.bmp</normaloff>:/icons/images/100.bmp</iconset>
   </property>
   <property name="text">
    <string>新建</string>
   </property>
  </action>
  <action name="actFile_Open">
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/001.GIF</normaloff>:/icons/images/001.GIF</iconset>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
  </action>
  <action name="actFile_Save">
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/104.bmp</normaloff>:/icons/images/104.bmp</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
  </action>
  <action name="actClose">
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/132.bmp</normaloff>:/icons/images/132.bmp</iconset>
   </property>
   <property name="text">
    <string>关闭</string>
   </property>
  </action>
  <action name="actEdit_Cut">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/200.bmp</normaloff>:/icons/images/200.bmp</iconset>
   </property>
   <property name="text">
    <string>剪切</string>
   </property>
  </action>
  <action name="actEdit_Copy">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/202.bmp</normaloff>:/icons/images/202.bmp</iconset>
   </property>
   <property name="text">
    <string>复制</string>
   </property>
  </action>
  <action name="actEdit_Paste">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/204.bmp</normaloff>:/icons/images/204.bmp</iconset>
   </property>
   <property name="text">
    <string>粘贴</string>
   </property>
  </action>
  <action name="actEdit_Undo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/206.bmp</normaloff>:/icons/images/206.bmp</iconset>
   </property>
   <property name="text">
    <string>撤销</string>
   </property>
  </action>
  <action name="actEdit_Redo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/208.bmp</normaloff>:/icons/images/208.bmp</iconset>
   </property>
   <property name="text">
    <string>重做</string>
   </property>
  </action>
  <action name="actEdit_SelectAll">
   <property name="text">
    <string>全选</string>
   </property>
  </action>
  <action name="actEdit_Clear">
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/212.bmp</normaloff>:/icons/images/212.bmp</iconset>
   </property>
   <property name="text">
    <string>清空</string>
   </property>
  </action>
  <action name="actFont_Bold">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/500.bmp</normaloff>:/icons/images/500.bmp</iconset>
   </property>
   <property name="text">
    <string>粗体</string>
   </property>
  </action>
  <action name="actFont_Italic">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/502.bmp</normaloff>:/icons/images/502.bmp</iconset>
   </property>
   <property name="text">
    <string>斜体</string>
   </property>
  </action>
  <action name="actFont_UnderLine">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/504.bmp</normaloff>:/icons/images/504.bmp</iconset>
   </property>
   <property name="text">
    <string>下划线</string>
   </property>
  </action>
  <action name="actSys_ToggleText">
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/322.bmp</normaloff>:/icons/images/322.bmp</iconset>
   </property>
   <property name="text">
    <string>显示按钮文字</string>
   </property>
  </action>
  <action name="actLang_CN">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/CN.jpg</normaloff>:/icons/images/CN.jpg</iconset>
   </property>
   <property name="text">
    <string>汉语</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actLang_EN">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../Resource/res.qrc">
     <normaloff>:/icons/images/timg2.jpg</normaloff>:/icons/images/timg2.jpg</iconset>
   </property>
   <property name="text">
    <string>英语</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../Resource/res.qrc"/>
 </resources>
 <connections/>
</ui>
